src/map/status.cpp

```diff
@@ -12350,8 +12338,7 @@ int status_change_end_(struct block_list* bl, enum sc_type type, int tid, const
 				// "Ugly workaround"  [Skotlex]
 				// delays status change ending so that a skill that sets opt1 fails to
 				// trigger when it also removed one
-				case SC_STONE:
-					sce->val4 = -1; // Petrify time
+				case SC_STONEWAIT:
```

Comment:
Instead of the ugly workaround we could/should add all those to the bottom of the function?